### ISSUE TITLE
Question: How to use SCHEDULE command class?

### ISSUE THREAD

**Issue by fondencn** [roblans/ZWave4Net#35]
Hello, 

I'd like to know How the SCHEDULE (0x53) zwave command class could be used via this library. 
Background: scheduling of heating Information for fibaro heat Controller thermostatic head. 

I did not find this command class in Zwave4Net's source code. Perhaps I'm on the wrong track, but it seems like I have to use this command class in order to program the heat Controller. 

See the device manual at
https://manuals.fibaro.com/content/manuals/en/FGT-001/FGT-001-EN-T-v1.0.pdf
Page 17ff.

**Comment by roblans** [comment]
The Schedule command class is not yet implemented.  You can implement it yourself, the specification of this command class is defined in this document:
[https://www.silabs.com/documents/login/miscellaneous/SDS13781-Z-Wave-Application-Command-Class-Specification.pdf](url)

**Comment by fondencn** [comment]
OK Thanks I'll try this as soon as possible and will push the enhancements.

**Comment by blakharaz** [comment]
A bit of necromancy...

I own a Fibaro FGT-001 myself and I dabbled a bit with the Schedule CC.
I ran into the problem that the responses can span over multiple message. Is there some mechanism to send a command and wait to receive multiple messages as response? A very ugly hack is here: https://github.com/blakharaz/ZWave4Net/blob/1f315ef84959cf020b2da5755b104c2e5e05b2b0/ZWave/CommandClasses/Schedule.cs#L104

**Comment by roblans** [comment]
There is no build in support for waiting on multiple response. You can use a TaskCompletionSource to accomplish this:

```cs
        public async Task<ScheduleStateReport> GetState(CancellationToken cancellationToken)
        {
            var completion = new TaskCompletionSource<ScheduleStateReport>();
            var responses = new List<byte[]>();

            void onEventReceived(object sender, NodeEventArgs args)
            {
                if (args.Command.ClassID == (byte)CommandClass.Schedule && args.Command.CommandID == (byte)command.StateReport)
                {
                    responses.Add(args.Command.Payload);
                    int numberOfFollowupReports = args.Command.Payload[1] >> 1;
                    if (numberOfFollowupReports == 0)
                    {
                        var report = new ScheduleStateReport(Node, responses);
                        completion.SetResult(report);
                    }
                }
            }

            Channel.NodeEventReceived += onEventReceived;
            try
            {
                await Send(new Command(Class, command.StateGet), cancellationToken);
                return await completion.Task;
            }
            finally
            {
                Channel.NodeEventReceived -= onEventReceived;
            }
        }
```

**Comment by fondencn** [comment]
I think this is fixed in PR #44 . Pulled and testet with my FGT-001 and Aeotec Thermostat, so I will close this issue. Thanks again.
